src/neuromaps_nhp/transforms.py

```diff
@@ -26,35 +26,67 @@ def _extract_res(nii_file: Path) -> tuple[float, float, float]:
     header = cast(Nifti1Header, img.header)
     return header.get_zooms()[:3]
 
-
-def _vol_to_vol(source: Path, target: Path) -> Path:
+def _vol_to_vol(
+    source: Path,
+    target: Path,
+    interp: str,
+    label: Path | None = None,) -> Path:
     """Transform a volumetric image from source space to target space.
 
     Args:
         source: Path to the source NIfTI volume to be transformed.
         target: Path to the target NIfTI volume defining the reference space.
+        interp: Interpolation method to use.
+        label: Optional path to a label image (optional for multiLabel).
 
     Returns:
         Path to the transformed NIfTI file written to disk.
-    """
-    # Later: include in the networkx graph, fetch and assert that the path exists.
 
+    Raises:
+        ValueError: unsupported interpolator.
     """
-    Can add a warning in the future if we want to allow for volumes to be resampled 
-    back to the original resolution in the target space, but generally, we would expect 
-    the output to be in the resolution of the target image. Given we are transforming 
-    to the target space, the warning is left out for now.
-    """
+    INTERP_PARAMS = {
+        "linear": ants.ants_apply_transforms_linear_params,
+        "nearestNeighbor": ants.ants_apply_transforms_nearest_neighbor_params,
+        "multiLabel": ants.ants_apply_transforms_multi_label_params,
+        "gaussian": ants.ants_apply_transforms_gaussian_params,
+        "BSpline": ants.ants_apply_transforms_bspline_params,
+        "cosineWindowedSinc": ants.ants_apply_transforms_cosine_windowed_sinc_params,
+        "welchWindowedSinc": ants.ants_apply_transforms_welch_windowed_sinc_params,
+        "hammingWindowedSinc": ants.ants_apply_transforms_hamming_windowed_sinc_params,
+        "lanczosWindowedSinc": ants.ants_apply_transforms_lanczos_windowed_sinc_params,
+    }
+
+    if interp not in INTERP_PARAMS:
+        raise ValueError("Unsupported interpolator.")
 
     out_file = target.parent / f"{source.stem}_to_{target.stem}.nii.gz"
-    interp = ants.ants_apply_transforms_linear_params()
+
+    # Default to linear if not specified
+    if not interp:
+        interp = "linear"
```

Comment:
One thing you can do is to set the default is replace these lines with passing "linear" in the method definition.

```python
def _vol_to_vol(
    source: Path,
    target: Path,
    interp: str = "linear",
    label: Path | None = None
) -> Path:
```